server/controllers/expenses.js

```diff
@@ -123,15 +125,29 @@ module.exports = (app) => {
     assertExpenseStatus(expense, status.PENDING)
       .then(() => {
         if (req.required.approved === false) {
-          return req.expense.setRejected()
-            .tap(expense => createActivity(expense, activities.GROUP_EXPENSE_REJECTED))
+          return expense.setRejected()
+            .tap(exp => createActivity(exp, activities.GROUP_EXPENSE_REJECTED))
         }
-        return fetchPaymentMethod(req.remoteUser.id)
+        if (expense.payoutMethod === 'manual') {
+          return getBalance(expense.GroupId)
+            .then(checkIfEnoughFunds(expense))
+            .then(() => expense.setApproved())
+            .tap(expense => createActivity(expense, activities.GROUP_EXPENSE_APPROVED))
```

Comment:
lines 133-135 and 144-146 are duplicate, can be extracted to end of promise chain at line 152